lib/addressable/template.rb

```diff
@@ -922,6 +922,7 @@ def normalize_value(value)
     # @return [Hash]
     #   A hash with stringified keys
     def normalize_keys(mapping)
+      return mapping unless mapping.respond_to?(:inject)
```

Comment:
As I was adding more tests/docs for this I realized it's incomplete for `#partial_expand` which requires something responding to `#key?` because nil is a valid expansion value (which becomes an empty string)